src/server/rdb_save.cc

```diff
@@ -743,6 +743,9 @@ class RdbSaver::Impl {
 
   error_code ConsumeChannel();
 
+  // Manually close channel.
+  void CloseChannel();
+
```

Comment:
Forgot to remove this... If we have no producers (header case), the channel should be closed on start (I've checked the internals). See the DCHECK below on github